net-p2p/bitcoin-core/bitcoin-core-25.0.ebuild

```diff
@@ -0,0 +1,251 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit autotools bash-completion-r1 db-use desktop python-any-r1 systemd xdg-utils
+
+DESCRIPTION="Reference implementation of the Bitcoin cryptocurrency"
+HOMEPAGE="https://bitcoincore.org/"
+SRC_URI="https://bitcoincore.org/bin/${P}/${P/-core}.tar.gz"
+S="${WORKDIR}/${P/-core}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+# IUSE="+cli" doesn't work due to https://bugs.gentoo.org/831045#c3
+IUSE="+asm +berkdb +bitcoin-cli +daemon dbus examples +external-signer kde libs +man nat-pmp +qrcode qt5 +sqlite system-leveldb +system-libsecp256k1 systemtap test upnp zeromq"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	dbus? ( qt5 )
+	kde? ( qt5 )
+	qrcode? ( qt5 )
+	system-leveldb? ( || ( daemon qt5 ) )
+"
+# dev-libs/univalue is now bundled, as upstream dropped support for system copy
+# and their version in the Bitcoin repo has deviated a fair bit from upstream.
+# Upstream also seems very inactive.
+RDEPEND="
+	!<dev-util/bitcoin-tx-25.0
+	>=dev-libs/boost-1.81.0:=
+	>=dev-libs/libevent-2.1.12:=
+	berkdb? ( >=sys-libs/db-4.8.30:$(db_ver_to_slot 4.8)=[cxx] )
+	bitcoin-cli? ( !<net-p2p/bitcoin-cli-25.0 )
+	daemon? (
+		!<net-p2p/bitcoind-25.0
+		acct-group/bitcoin
+		acct-user/bitcoin
+	)
+	libs? ( !<net-libs/libbitcoinconsensus-25.0 )
+	nat-pmp? ( >=net-libs/libnatpmp-20220705:= )
+	qrcode? ( >=media-gfx/qrencode-4.1.1:= )
+	qt5? (
+		!<net-p2p/bitcoin-qt-25.0
+		>=dev-qt/qtcore-5.15.5:5
+		>=dev-qt/qtgui-5.15.5:5
+		>=dev-qt/qtnetwork-5.15.5:5
+		>=dev-qt/qtwidgets-5.15.5:5
+		dbus? ( >=dev-qt/qtdbus-5.15.5:5 )
+	)
+	sqlite? ( >=dev-db/sqlite-3.38.5:= )
+	system-leveldb? ( virtual/bitcoin-leveldb )
+	system-libsecp256k1? ( >=dev-libs/libsecp256k1-0.3.1:=[recovery,schnorr] )
+	upnp? ( >=net-libs/miniupnpc-2.2.2:= )
+	zeromq? ( >=net-libs/zeromq-4.3.4:= )
+"
+DEPEND="
+	${RDEPEND}
+	systemtap? ( >=dev-util/systemtap-4.8 )
+"
+BDEPEND="
+	virtual/pkgconfig
+	daemon? (
+		acct-group/bitcoin
+		acct-user/bitcoin
+	)
+	qt5? ( >=dev-qt/linguist-tools-5.15.5:5 )
+	test? ( ${PYTHON_DEPS} )
+"
+IDEPEND="
+	qt5? ( dev-util/desktop-file-utils )
+"
+
+DOCS=(
+	doc/bips.md
+	doc/bitcoin-conf.md
+	doc/descriptors.md
+	doc/files.md
+	doc/i2p.md
+	doc/JSON-RPC-interface.md
+	doc/multisig-tutorial.md
+	doc/p2p-bad-ports.md
+	doc/psbt.md
+	doc/reduce-memory.md
+	doc/reduce-traffic.md
+	doc/release-notes.md
+	doc/REST-interface.md
+	doc/tor.md
+)
+
+PATCHES=(
+	"${FILESDIR}/25.0-syslibs.patch"
+	"${FILESDIR}/init.patch"
+)
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	! use system-leveldb || rm -r src/leveldb || die
+	if use system-libsecp256k1 ; then
```

Comment:
I always put spaces around shell operators to make it very clear that they're not part of any argument/word. Without the leading space, it could be easier to believe that the argument passed to the `use` function is `system-libsecp256k1;`.